### PR TITLE
Add decomp for max_pool1d

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89506
* #89434

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -2237,6 +2237,96 @@ def binary_cross_entropy_with_logits(
...
+            stride,
+        )
+        + 1
+    )
+    if ceil_mode:
+        if (outputSize - 1) * stride >= inputSize + pad_l:
+            outputSize -= 1
+    return outputSize
+
+# copied from _meta_registrations (TODO: create a common utils file)
```
Comment (anjali411) [inline torch/_decomp/decompositions.py:2269]: as per this comment, need to move all the helper functions to common utils before merging

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -2237,6 +2237,96 @@ def binary_cross_entropy_with_logits(
...
     return apply_loss_reduction(loss, reduction)
 
+def div_rtn(x, y):
+    q = x // y
+    r = x % y
+    # WARNING: explicit bool conversion here is necessary;
+    # would be fixed by SymBool
+    if r != 0 and (bool(r < 0) != bool(y < 0)):
+        q -= 1
+    return q
```
Comment (ezyang) [inline torch/_decomp/decompositions.py:2247]: we already have a copy of this in meta_registrations

File: torch/_decomp/decompositions.py [hunk 3]
```diff
@@ -2237,6 +2237,96 @@ def binary_cross_entropy_with_logits(
...
+
+def pooling_output_shape_pad_lr(
```
Comment (ezyang) [inline torch/_decomp/decompositions.py:2249]: this is copied too, I guess?

Comment (anjali411) [reply torch/_decomp/decompositions.py:2249]: yeah please see comments above

**Pull request closed without merging**